gadget/raw.go

```diff
@@ -257,18 +260,78 @@ func (r *rawStructureUpdater) rollbackDifferent(out io.WriteSeeker, pc *LaidOutC
 	return nil
 }
 
+var eMMCDeviceRegex = regexp.MustCompile("mmcblk[0-9]boot[0-1]")
+
+// setEMMCPartitionReadWrite changes the read-write status of a eMMC boot partition.
+// Documentation for force_ro here:
+// https://www.kernel.org/doc/Documentation/mmc/mmc-dev-attrs.txt
+var setEMMCPartitionReadWrite = func(device string, rw bool) error {
+	sdevPath := fmt.Sprintf("/sys/block/%s/force_ro", device)
+
+	f, err := os.OpenFile(path.Join(dirs.GlobalRootDir, sdevPath), os.O_WRONLY, 0)
+	if err != nil {
+		return fmt.Errorf("cannot open device for writing: %v", err)
+	}
+	defer f.Close()
+
+	d := "1"
+	if rw {
+		d = "0"
+	}
+	if _, err := f.Write([]byte(d)); err != nil {
+		return fmt.Errorf("cannot change device read only mode to %v: %w", d, err)
+	}
+	return nil
+}
+
+func openDiskForWrite(device string, laidOutStruct *LaidOutStructure) (*os.File, func(), error) {
```

Comment:
sorry I missed that earlier. Technically os.File.Close and io.Closer.Close() return an error. If we're returning an equivalent of those, perhaps it should return an error as well

```suggestion
func openDiskForWrite(device string, laidOutStruct *LaidOutStructure) (*os.File, func() error, error) {
```